contracts/Rentals.sol

```diff
@@ -396,19 +393,37 @@ contract Rentals is NonceVerifiable, NativeMetaTransaction, IERC721Receiver {
 
         Rental storage rental = rentals[_contractAddress][_tokenId];
 
-        if (rental.lessor != address(0)) {
-            // The contract already has the asset, so we just need to validate that the original owner matches the provided lessor.
+        // True if the asset is currently rented.
+        bool rented = isRented(_contractAddress, _tokenId);
+        // True if the asset rental period is over, but is has not been claimed back from the contract.
+        bool reRent = !rented && rental.lessor != address(0);
+        // True if the asset rental period is not over yet, but the lessor and the tenant are the same.
+        bool extend = rented && rental.lessor == _lessor && rental.tenant == _tenant;
+
+        if (!extend && !reRent) {
+            // Verify that the asset is not already rented.
+            require(!rented, "Rentals#_rent: CURRENTLY_RENTED");
```

Comment:
What do you think of moving the if into something like

```
require(extend || reRent ||  !rented, "Rentals#_rent: CURRENTLY_RENTED");
```

consume less gas?

Reply:
It's kind of complicated to read